### DISCUSSION TITLE
Question about setting up local development environment

### DISCUSSION THREAD

**Discussion by sidney-eliot** [gotson/komga#1801] [Q&A]
I would be interested in contributing to Komga, but I'm having some issues with getting Gradle set up locally. For some context, I have little to none experience with the frameworks Komga uses, but do have dev experience with other frameworks.

I closely followed the DEVELOPING.md file, and the things I did so far are:

- run "npm install" inside the "komga-webui" folder
- run "gradlew tasks" (which is a nice overview of all available tasks)
- run "gradlew build"
- run  "gradlew prepareThymeLeaf"
- run "gradlew  generateJooq" & "gradlew generateTasksJooq"
- run "gradlew bootRun --args='--spring.profiles.active=dev'" (which is supposed to run the application locally using the dev profile. **But it indefinitely stalls at 95% and says. "Try to process first available task" as well as "No available task found"**)

What I've also figured out is that to run the UI one uses "npm run serve" from within the "komga-webui" folder and in a separate terminal runs the above-mentioned commands, to supply the frontend with a backend. Komga is then available at "localhost:8081" with the login of U=admin@example.org, P=admin.

What I haven't figured out how to run yet, is "jooq-codegen-primary", which might very likely be the cause of my issues.

I would appreciate a tip on what I'm doing wrong here, it's probably something obvious I'm overlooking.

**Comment by gotson** [comment]
hi, can you maybe provide some logs of the `gradlew bootRun` ?
